### PR TITLE
Switched `Locale` deprecation maps to jumptables and fixed minor bug with deprecated countries

### PR DESCRIPTION
**Pull request by gaaclarke**
1) Fixed bug with `Locale` where deprecated countries would return all the same data, but the would not be considered `==`
2) Switched the `const Map` data representing the deprecated languages and countries to jump tables.  In microbenchmarks it behaves 2x faster in jit x64.
3) I reordered the `==` operator to prefer less lookup calls to deprecated tables.

I made this change because `Locale` is sometimes used as a key into a map, as it is in Flutter gallery.  When it is used in a Map it's `hashCode` is calculated, which is based on the country code and the language code, which requires lookups into these maps.  I tried to memoize the values and the hashCode, but because of the limitations of Dart I couldn't do that (https://github.com/dart-lang/sdk/issues/48948).

The reason I identified this as a potential performance fix is that I setup Flutter to continuously build frames and navigated through the Material section of Flutter gallery and noticed that `Locale.hashCode` shows up pretty high.  I also saw other Flutter users in Google using Locale as a key.
Screen Shot 2022-05-03 at 10 47 11 AM

Note: 
1) I don't expect this to make a difference in `customer: money`.
1) While this is verbose code, it should actually compile to smaller binary size.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/ui/platform_dispatcher.dart [hunk 1]
```diff
@@ -1768,6 +1768,852 @@ enum DisplayFeatureState {
...
+        }
+        break;
+      }
+  }
+  return x;
+}
+
+// This map is generated by //flutter/tools/gen_locale.dart
+// Mappings generated for language subtag registry as of 2019-02-27.
+// static const Map<String, String> _deprecatedLanguageSubtagMap = <String, String>{
```
Comment (dnfield) [inline lib/ui/platform_dispatcher.dart:1844]: nit: add something about this map being a more readable version of the jump table below.

File: lib/ui/platform_dispatcher.dart [hunk 2]
```diff
@@ -1768,6 +1768,852 @@ enum DisplayFeatureState {
...
+          case 82 /*R*/ :
+            return 'CD';
+        }
+        break;
+      }
+  }
+  return x;
+}
+
+// This map is generated by //flutter/tools/gen_locale.dart
```
Comment (dnfield) [inline lib/ui/platform_dispatcher.dart:1842]: We should probably replace that file with your gist, no?

**Review by dnfield — COMMENTED** [review]
Is this fater in AOT or just JIT?

**Comment by gaaclarke** [comment]
@dnfield, slava pointed out that we don't actually compile to jump tables with Dart, so the performance isn't always going to be faster than a map.  I can switch it to a proper jump table like `lex` does but it's a bit harder to read and know it's correct.  I'm on the fence about wether it's worth doing.  I'm going to pull this out of review and think about it.

**Comment by gaaclarke** [comment]
Abandoned for the alternative: https://github.com/flutter/engine/pull/33140

**Pull request closed without merging**